proxy/proxy.go

```diff
@@ -882,7 +884,31 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 
 	p.metrics.IncCounter("outgoing." + req.Proto)
 	ctx.proxySpan.LogKV("http_roundtrip", StartEvent)
-	response, err := p.roundTripper.RoundTrip(req)
+
+	var response *http.Response
+	switch req.URL.Scheme {
+	case "fastcgi":
+		f := "index.php"
+		if sf, ok := ctx.StateBag()["fastCgiFilename"]; ok {
+			f = sf.(string)
```

Comment:
it would be safer to check whether it is a string, because this way a buggy filter can set something else in the state bag.

Reply:
discussed: failing fast via crashing is also fine in this case